### ISSUE TITLE
Marathon fails to upgrade 1.1.1 -> 1.3.0

### ISSUE THREAD

**Issue by bobrik** [mesosphere/marathon#4470]
@alberts, I blame you for this app :)

```
2016-09-23T16:44:44.095 foobar [2016-09-23 16:44:44,094] INFO Elected (LeaderLatchListener Interface) (mesosphere.marathon.core.election.impl.CuratorElectionService:ForkJoinPool-2-worker-29-EventThread)
2016-09-23T16:44:44.095 foobar [2016-09-23 16:44:44,094] INFO As new leader running the driver (mesosphere.marathon.MarathonSchedulerService$$EnhancerByGuice$$46c953c5:ForkJoinPool-2-worker-29-EventThread)
2016-09-23T16:44:44.098 foobar [2016-09-23 16:44:44,097] INFO Migration for storage: Version(1, 1, 1) to current: Version(1, 3, 0): apply change for version: Version(1, 2, 0)  (mesosphere.marathon.state.Migration:ForkJoinPool-2-worker-29)
2016-09-23T16:44:44.098 foobar [2016-09-23 16:44:44,097] INFO Start 1.2 migration (mesosphere.marathon.state.MigrationTo1_2:ForkJoinPool-2-worker-71)
2016-09-23T16:44:44.104 foobar [2016-09-23 16:44:44,103] INFO Discovered 55 tasks for status migration (mesosphere.marathon.state.MigrationTo1_2:ForkJoinPool-2-worker-29)
2016-09-23T16:44:44.109 foobar [2016-09-23 16:44:44,108] ERROR Failed to take over leadership (mesosphere.marathon.core.election.impl.CuratorElectionService:ForkJoinPool-2-worker-29-EventThread)
2016-09-23T16:44:44.109 foobar mesosphere.marathon.MigrationFailedException: while migrating storage to 1.2
2016-09-23T16:44:44.109 foobar 	at mesosphere.marathon.state.Migration$$anonfun$migrations$5$$anonfun$apply$4.applyOrElse(Migration.scala:64)
2016-09-23T16:44:44.110 foobar 	at mesosphere.marathon.state.Migration$$anonfun$migrations$5$$anonfun$apply$4.applyOrElse(Migration.scala:63)
2016-09-23T16:44:44.110 foobar 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
2016-09-23T16:44:44.110 foobar 	at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216)
2016-09-23T16:44:44.110 foobar 	at scala.util.Try$.apply(Try.scala:192)
2016-09-23T16:44:44.110 foobar 	at scala.util.Failure.recover(Try.scala:216)
2016-09-23T16:44:44.111 foobar 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-09-23T16:44:44.111 foobar 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-09-23T16:44:44.111 foobar 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
2016-09-23T16:44:44.111 foobar 	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
2016-09-23T16:44:44.112 foobar 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
2016-09-23T16:44:44.112 foobar 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
2016-09-23T16:44:44.112 foobar 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
2016-09-23T16:44:44.112 foobar 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
2016-09-23T16:44:44.112 foobar Caused by: mesosphere.marathon.StoreCommandFailedException: Could not fetch MarathonTaskState with key: pdx_druid_main_broker:pdx_druid_main_broker.bc9607ec-70c7-11e5-8d50-2c600c6fb621
2016-09-23T16:44:44.113 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$mesosphere$marathon$state$MarathonStore$$exceptionTransform$1.applyOrElse(MarathonStore.scala:84)
2016-09-23T16:44:44.113 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$mesosphere$marathon$state$MarathonStore$$exceptionTransform$1.applyOrElse(MarathonStore.scala:83)
2016-09-23T16:44:44.113 foobar 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
2016-09-23T16:44:44.113 foobar 	at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216)
2016-09-23T16:44:44.114 foobar 	at scala.util.Try$.apply(Try.scala:192)
2016-09-23T16:44:44.114 foobar 	at scala.util.Failure.recover(Try.scala:216)
2016-09-23T16:44:44.114 foobar 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-09-23T16:44:44.114 foobar 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-09-23T16:44:44.114 foobar 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
2016-09-23T16:44:44.115 foobar 	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
2016-09-23T16:44:44.115 foobar 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
2016-09-23T16:44:44.115 foobar 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
2016-09-23T16:44:44.115 foobar 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
2016-09-23T16:44:44.115 foobar 	... 2 common frames omitted
2016-09-23T16:44:44.116 foobar Caused by: com.google.protobuf.InvalidProtocolBufferException: Protocol message end-group tag did not match expected tag.
2016-09-23T16:44:44.116 foobar 	at com.google.protobuf.InvalidProtocolBufferException.invalidEndTag(InvalidProtocolBufferException.java:94)
2016-09-23T16:44:44.116 foobar 	at com.google.protobuf.CodedInputStream.checkLastTagWas(CodedInputStream.java:174)
2016-09-23T16:44:44.116 foobar 	at com.google.protobuf.AbstractParser.parsePartialFrom(AbstractParser.java:139)
2016-09-23T16:44:44.117 foobar 	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:168)
2016-09-23T16:44:44.117 foobar 	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:180)
2016-09-23T16:44:44.117 foobar 	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:185)
2016-09-23T16:44:44.117 foobar 	at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:49)
2016-09-23T16:44:44.117 foobar 	at mesosphere.marathon.Protos$MarathonTask.parseFrom(Protos.java:16308)
2016-09-23T16:44:44.118 foobar 	at mesosphere.marathon.state.MarathonTaskState.mergeFromProto(MarathonTaskState.scala:15)
2016-09-23T16:44:44.118 foobar 	at mesosphere.marathon.state.MarathonTaskState.mergeFromProto(MarathonTaskState.scala:9)
2016-09-23T16:44:44.118 foobar 	at mesosphere.marathon.state.MarathonStore.mesosphere$marathon$state$MarathonStore$$stateFromBytes(MarathonStore.scala:88)
2016-09-23T16:44:44.119 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$fetch$1$$anonfun$apply$2.apply(MarathonStore.scala:36)
2016-09-23T16:44:44.119 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$fetch$1$$anonfun$apply$2.apply(MarathonStore.scala:34)
2016-09-23T16:44:44.119 foobar 	at scala.Option.map(Option.scala:146)
2016-09-23T16:44:44.119 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$fetch$1.apply(MarathonStore.scala:34)
2016-09-23T16:44:44.119 foobar 	at mesosphere.marathon.state.MarathonStore$$anonfun$fetch$1.apply(MarathonStore.scala:34)
2016-09-23T16:44:44.120 foobar 	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
2016-09-23T16:44:44.120 foobar 	at scala.util.Try$.apply(Try.scala:192)
2016-09-23T16:44:44.120 foobar 	at scala.util.Success.map(Try.scala:237)
2016-09-23T16:44:44.120 foobar 	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
2016-09-23T16:44:44.121 foobar 	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
2016-09-23T16:44:44.121 foobar 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
2016-09-23T16:44:44.121 foobar 	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
2016-09-23T16:44:44.121 foobar 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
2016-09-23T16:44:44.121 foobar 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
2016-09-23T16:44:44.122 foobar 	... 2 common frames omitted
2016-09-23T16:44:44.122 foobar [2016-09-23 16:44:44,108] INFO Abdicating leadership while leading (reoffer=false) (mesosphere.marathon.core.election.impl.CuratorElectionService:ForkJoinPool-2-worker-29-EventThread)
```

**Comment by bobrik** [comment]
I don't even have `/pdx/druid/main/broker`. The closest thing is `/pdx/druid/svc`.

**Comment by bobrik** [comment]
I nuked `/pdx/druid` group, but that didn't help.

**Comment by bobrik** [comment]
Looking at znode `state/task:pdx_druid_main_broker:pdx_druid_main_broker.bc9607ec-70c7-11e5-8d50-2c600c6fb621`:

```
{
    "Id": "8ab87a7b91a410d3a176de89a8df66ce98bb627deb6c3edd2e35750d5645cb69",
    "Created": "2015-10-12T09:58:10.409111868Z",
```

Nothing from 2015 survived on this Marathon. My guess is that Marathon forgot about this znode a few versions ago, but now it rediscovered it again when it can't migrate it nicely. It'd be nice to deal with it gracefully somehow, but I'm not sure how.

**Comment by jasongilanfarr** [comment]
In Migration1_2, we should probably warn if we were unable to load a task due to a Protobuf error.

**Comment by jasongilanfarr** [comment]
@zen-dog - please close when the referenced PR is in the 1.3 branch

**Comment by bobrik** [comment]
The error is still there on 1.3.1:

```
2016-10-06T09:54:10.325 myhost [2016-10-06 09:54:10,324] ERROR Failed to take over leadership (mesosphere.marathon.core.election.impl.CuratorElectionService:ForkJoinPool-2-worker-29-EventThread)
2016-10-06T09:54:10.325 myhost mesosphere.marathon.MigrationFailedException: while migrating storage to 1.2
2016-10-06T09:54:10.325 myhost 	at mesosphere.marathon.state.Migration$$anonfun$migrations$5$$anonfun$apply$4.applyOrElse(Migration.scala:64)
2016-10-06T09:54:10.326 myhost 	at mesosphere.marathon.state.Migration$$anonfun$migrations$5$$anonfun$apply$4.applyOrElse(Migration.scala:63)
2016-10-06T09:54:10.326 myhost 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
2016-10-06T09:54:10.326 myhost 	at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:216)
2016-10-06T09:54:10.326 myhost 	at scala.util.Try$.apply(Try.scala:192)
2016-10-06T09:54:10.327 myhost 	at scala.util.Failure.recover(Try.scala:216)
2016-10-06T09:54:10.327 myhost 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-10-06T09:54:10.327 myhost 	at scala.concurrent.Future$$anonfun$recover$1.apply(Future.scala:326)
2016-10-06T09:54:10.327 myhost 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
2016-10-06T09:54:10.327 myhost 	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
2016-10-06T09:54:10.327 myhost 	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
2016-10-06T09:54:10.328 myhost 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1253)
2016-10-06T09:54:10.328 myhost 	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1346)
2016-10-06T09:54:10.328 myhost 	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
2016-10-06T09:54:10.328 myhost 	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
2016-10-06T09:54:10.328 myhost Caused by: mesosphere.marathon.MigrationFailedException: Inconsistency in the task store detected, task with id pdx_druid_main_broker:pdx_druid_main_broker.bc9607ec-70c7-11e5-8d50-2c600c6fb621 not found, but delivered in allIds().
2016-10-06T09:54:10.329 myhost 	at mesosphere.marathon.state.MigrationTo1_2$stateMachine$macro$34$1$$anonfun$loadAndMigrateTasks$macro$43$1.apply(Migration.scala:409)
2016-10-06T09:54:10.329 myhost 	at mesosphere.marathon.state.MigrationTo1_2$stateMachine$macro$34$1$$anonfun$loadAndMigrateTasks$macro$43$1.apply(Migration.scala:399)
2016-10-06T09:54:10.329 myhost 	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:253)
2016-10-06T09:54:10.329 myhost 	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:251)
2016-10-06T09:54:10.329 myhost 	... 7 common frames omitted
```

**Comment by bobrik** [comment]
Still an issue in 1.3.3.

**Comment by jasongilanfarr** [comment]
Fixing in https://phabricator.mesosphere.com/D114

**Comment by jasongilanfarr** [comment]
This should be fixed in the upcoming 1.3.4 (waiting on one more patch to release it)

**Comment by jdef** [comment]
should be fixed in the newly minted 1.3.5, please re-open if still broken for you

**Comment by bobrik** [comment]
It seems to be working, thanks!
